packages/terra-table/src/subcomponents/ColumnHeader.jsx

```diff
@@ -86,6 +86,7 @@ const ColumnHeader = (props) => {
       <tr
         aria-rowindex={1}
         className={cx('column-header-row', { hidden: !hasColumnHeaders })}
+        data-row-id="table-header-row"
```

Comment:
Would not be a correct row id.  If you had two Tables on the page, you would have the same id and the DOM would be broken.